Classes/DataProcessing/RootSiteProcessing/SiteProvider.php

```diff
@@ -149,7 +148,7 @@ private function filterSites(array $allowedSites = []): array
 
         foreach ($allSites as $site) {
             if (in_array($site->getRootPageId(), $allowedSites, true) &&
-                $site->getConfiguration()['headless'] ?? false) {
+            $site->getConfiguration()['headless'] ?? false) {
```

Comment:
Migrate to `GeneralUtility::makeInstance(HeadlessMode::class)->withRequest($request)->isEnabled()`